src/login/logind-session-dbus.c

```diff
@@ -413,6 +413,27 @@ static int method_set_type(sd_bus_message *message, void *userdata, sd_bus_error
         return sd_bus_reply_method_return(message, NULL);
 }
 
+static int method_set_display(sd_bus_message *message, void *userdata, sd_bus_error *error) {
+        Session *s = ASSERT_PTR(userdata);
+        const char *display;
+        int r;
+
+        assert(message);
+
+        r = sd_bus_message_read(message, "s", &display);
+        if (r < 0)
+                return r;
+
+        if (!session_is_controller(s, sd_bus_message_get_sender(message)))
+                return sd_bus_error_set(error, BUS_ERROR_NOT_IN_CONTROL, "You must be in control of this session to set display");
+
+        r = free_and_strdup(&s->display, display);
```

Comment:
Does the right thing happen if the session type is not a graphical session type?

Shouldn't this call be `session_set_display` or something and have it save the results?

Reply:
This needs to have the same logic as `method_set_type()` above, i.e. save the value and send a signal in addition to changing the internal value. We currently don't restart logind because of bugs, but we certainly want to, and this setting would be lost then.

Reply:
Updated. PTAL.